tests/Operation/WatchFunctionalTest.php

```diff
@@ -123,19 +124,17 @@ function (array $event) use (&$events) {
 
         (new CommandObserver())->observe(
             function () use ($changeStream) {
-                $changeStream->next();
+                $this->advanceCursorUntilValid($changeStream);
             },
             function (array $event) use (&$events) {
                 $events[] = $event;
             }
         );
 
-        $this->assertCount(1, $events);
-        $this->assertSame('getMore', $events[0]['started']->getCommandName());
-        $postBatchResumeToken = $this->getPostBatchResumeTokenFromReply($events[0]['succeeded']->getReply());
+        $lastEvent = array_pop($events);
```

Comment:
Alternatively, we could just swap `$events = [];` for `$lastEvent` and let the second callable overwrite the variable, but I'm fine sticking to the array for a minimal diff and consistency with other methods.

I might suggest a non-empty assertion before we call `array_pop()`, though.

Reply:
Good point. Not storing all events definitely makes sense in this test.